src/Features/Core/Portable/Completion/Providers/AbstractTypeImportCompletionProvider.cs

```diff
@@ -0,0 +1,322 @@
+
+            var finalText = root.GetText(text.Encoding)
+                .Replace(completionItem.Span, completionItem.DisplayText.Trim());
+
+            document = document.WithText(finalText);
+
+            tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            root = await tree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+            var addedNode = root.FindNode(completionItem.Span);
+            var annotatedNode = addedNode.WithAdditionalAnnotations(_annotation);
+
+            root = root.ReplaceNode(addedNode, annotatedNode);
+            document = document.WithSyntaxRoot(root);
+
+            if (completionItem is TypeImportCompletionItem importItem)
+            {
+                var optionSet = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+                var placeSystemNamespaceFirst = optionSet.GetOption(GenerationOptions.PlaceSystemNamespaceFirst, document.Project.Language);
+                var importNode = CreateImport(document, importItem.ContainingNamespace);
+
+                var addImportService = document.GetLanguageService<IAddImportsService>();
+                var compilation = await document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+
+                root = addImportService.AddImport(compilation, root, annotatedNode, importNode, placeSystemNamespaceFirst);
+                document = document.WithSyntaxRoot(root);
+
+                document = await Formatter.FormatAsync(document, Formatter.Annotation, cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+
+            return document;
+        }
+
+        private static SyntaxNode CreateImport(Document document, string namespaceName)
+        {
+            var syntaxGenerator = SyntaxGenerator.GetGenerator(document);
+            return syntaxGenerator.NamespaceImportDeclaration(namespaceName).WithAdditionalAnnotations(Formatter.Annotation);
+        }
+
+        protected override async Task<CompletionDescription> GetDescriptionWorkerAsync(
+            Document document, CompletionItem item, CancellationToken cancellationToken)
+        {
+            if (item is TypeImportCompletionItem importItem)
+            {
+                var compilation = await document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+                var symbol = compilation.GetTypeByMetadataName(importItem.MetadataName);
+                if (symbol != null)
+                {
+                    var semanticModel = await document.GetSemanticModelAsync().ConfigureAwait(false);
+#if DEBUG
+                    return CompletionDescription.FromText(DebugObject.DebugText);
+#else
+                    return await CommonCompletionUtilities.CreateDescriptionAsync(
+                        document.Project.Solution.Workspace,
+                        semanticModel,
+                        position: 0,
```

Comment:
what is this position used for?

Reply:
Not sure, but it's needed for creating description